src/main/java/com/aws/greengrass/componentmanager/ComponentManager.java

```diff
@@ -148,11 +147,24 @@ ComponentMetadata resolveComponentVersion(String componentName, Map<String, Requ
                     + " candidate as the resolved version without negotiating with cloud.");
             resolvedComponentId = localCandidateOptional.get();
         } else {
-            // otherwise try to negotiate with cloud
+            // Otherwise try to negotiate with cloud
             logger.atInfo().setEventType("negotiate-version-with-cloud-start").log("Negotiating version with cloud");
 
-            resolvedComponentId =
-                    negotiateVersionWithCloud(componentName, versionRequirements, localCandidateOptional.orElse(null));
+            if (deviceConfiguration.isDeviceConfiguredToTalkToCloud()) {
```

Comment:
Do we not need similar changes for artifacts download? Is that already covered?

Reply:
So I saw [here](https://docs.aws.amazon.com/greengrass/v2/developerguide/getting-started.html#create-first-component) the recipe for local deployment doesn't declare artifacts (just directly uses it). Is this always the case? Then we don't need change because artifacts will be empty.

Reply:
No, if the recipe lists artifacts then we might try to download them (if they don't exist already).

Reply:
Move this check above the log line, otherwise the log doesn't make sense since we're skipping.